common/src/main/java/org/keycloak/common/util/FindFile.java

```diff
@@ -29,28 +29,32 @@
  * @version $Revision: 1 $
  */
 public class FindFile {
-    public static InputStream findFile(String keycloakConfigFile) {
-        if (keycloakConfigFile.startsWith(GenericConstants.PROTOCOL_CLASSPATH)) {
-            String classPathLocation = keycloakConfigFile.replace(GenericConstants.PROTOCOL_CLASSPATH, "");
-            // Try current class classloader first
-            InputStream is = FindFile.class.getClassLoader().getResourceAsStream(classPathLocation);
-            if (is == null) {
-                is = Thread.currentThread().getContextClassLoader().getResourceAsStream(classPathLocation);
-            }
+	public static InputStream findFile(String keycloakConfigFile) {
+		if (keycloakConfigFile.startsWith(GenericConstants.PROTOCOL_CLASSPATH)) {
+			String classPathLocation = keycloakConfigFile.replace(GenericConstants.PROTOCOL_CLASSPATH, "");
+			// Try current class classloader first
+			InputStream is = FindFile.class.getClassLoader().getResourceAsStream(classPathLocation);
+			if (is == null) {
+				is = Thread.currentThread().getContextClassLoader().getResourceAsStream(classPathLocation);
+			}
 
-            if (is != null) {
-                return is;
-            } else {
-                throw new RuntimeException("Unable to find config from classpath: " + keycloakConfigFile);
-            }
-        } else {
-            // Fallback to file
-            try {
-            	//Added support for the URL handling
-                return new URL(keycloakConfigFile).openStream();
-            } catch (Exception e) {
-                throw new RuntimeException(e);
-            }
-        }
-    }
+			if (is != null) {
+				return is;
+			} else {
+				throw new RuntimeException("Unable to find config from classpath: " + keycloakConfigFile);
+			}
+		} else {
+			// Fallback to file
+			try {
+				// Added support for the URL handling
+				if (!keycloakConfigFile.contains(":")) {
+					// assume file protocol
```

Comment:
This class is for loading local resources, not going to the internet. Looking at this code - if I pass it "http://www.keycloak.org/" it will just go out and fetch it.

I see the need for custom resource loading using 'protocol:' but a better place to solve that would be in [AbstractKeycloakLoginModule](https://github.com/keycloak/keycloak/blob/3.3.0.CR2/adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/jaas/AbstractKeycloakLoginModule.java#L85-L94)

Reply:
I looked in to the resolveDeployment method of AbstractKeycloakLoginModule. It is using the class FindFile method findFile. Please correct me if I am wrong so, as per you I should update it to provide the inputstream via the URL openstream and not calling the FindFile method.

Reply:
I think I get it now. You want me add my custom resolver for resource having protocol in  AbstractKeycloakLoginModule. If I understood correctly I will update this class with a method to resolve the same.